src/write.c

```diff
@@ -180,6 +180,74 @@ static void avifEncoderItemAddMdatFixup(avifEncoderItem * item, const avifRWStre
     fixup->offset = avifRWStreamOffset(s);
 }
 
+// ---------------------------------------------------------------------------
+// avifItemPropertyDedup - Provides ipco deduplication
+
+typedef struct avifItemProperty
+{
+    uint8_t index;
+    size_t offset;
+    size_t size;
+} avifItemProperty;
+AVIF_ARRAY_DECLARE(avifItemPropertyArray, avifItemProperty, property);
+
+typedef struct avifItemPropertyDedup
+{
+    avifItemPropertyArray properties;
+    avifRWStream s;    // Temporary stream for each new property, checked against already-written boxes for deduplications
+    avifRWData buffer; // Temporary storage for 's'
```

Comment:
Optional: It seems that we should be able to write each new property to the output stream tentatively, without the need for the temporary stream.

Reply:
I agree that this would be a viable approach, but I prefer keeping the buffer separate.